subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -858,7 +857,8 @@ private DefaultConfiguration createCopy(Set<Dependency> dependencies, Set<Depend
         copiedConfiguration.description = description;
 
         copiedConfiguration.defaultDependencyActions = defaultDependencyActions;
-        copiedConfiguration.dependencyResolutionListeners = dependencyResolutionListeners;
+        copiedConfiguration.withDependencyActions = withDependencyActions;
```

Comment:
Actually - I wonder if it's ok that we just share this (and the above).
It might be unexpected that additional actions on any of the configurations (after copying) seep into the other one.

In fact, don't these get cleared upon resolution too, meaning they have the same bug that this PR fixed for `dependencyResolutionListeners`? :(